plugins/pulp_python/plugins/importers/sync.py

```diff
@@ -23,11 +25,18 @@
 
 class DownloadMetadataStep(publish_step.DownloadStep):
     """
-    This DownloadStep subclass contains the code to process the downloaded manifests and decide what
-    to download from the feed. It does this as it gets each metadata file to spread the load on the
-    database.
+    This DownloadStep subclass contains the code to process the downloaded manifests and determine
+    what is available from the feed. It does this as it gets each metadata file to spread the load
+    on the database.
     """
 
+    def __init__(self, *args, **kwargs):
+        """
+        Injects download requests into the call to the parent class's __init__
+        """
+        kwargs['downloads'] = self.generate_download_requests()
```

Comment:
On Thu, Jan 14, 2016 at 06:23:54AM -0800, Michael Hrivnak wrote:
> > +        kwargs['downloads'] = self.generate_download_requests()
> 
> ``downloads`` is not a required param. In fact if a caller did pass it in, we overwrite that value here.

Ugh, I really hope I only wrote that because I hadn't had enough coffee
yet. Speaking of which, is it… coffee time?

-- 
Randy Barlow
irc:   bowlofeggs